apollo-router/src/services/router/service.rs

```diff
@@ -860,11 +877,7 @@ pub(crate) fn process_vary_header(headers: &mut HeaderMap<HeaderValue>) {
 #[derive(Clone)]
 pub(crate) struct RouterCreator {
     pub(crate) supergraph_creator: Arc<SupergraphCreator>,
-    static_page: StaticPageLayer,
-    apq_layer: APQLayer,
-    pub(crate) persisted_query_layer: Arc<PersistedQueryLayer>,
-    query_analysis_layer: QueryAnalysisLayer,
-    batching: Batching,
+    sb: Arc<parking_lot::Mutex<router::BoxCloneService>>,
```

Comment:
@BrynCooke Maybe this could be just buffer rather than in a mutex.

Reply:
We can get rid of the mutex. It requires an extra box, either on the service after it's cloned or on the future. I don't have strong feeling here.

Reply:
done: 9b7542300e03fc18de7d0a2447703af322169a55